airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceConfiguration.kt

```diff
@@ -101,57 +106,35 @@ class MySqlSourceConfigurationFactory @Inject constructor(val featureFlags: Set<
-                    if (
-                        featureFlags.contains(FeatureFlag.AIRBYTE_CLOUD_DEPLOYMENT) &&
-                            sshTunnel is SshNoTunnelMethod
-                    ) {
-                        throw ConfigErrorException(
-                            "Connection from Airbyte Cloud requires " +
-                                "SSL encryption or an SSH tunnel."
-                        )
-                    }
-                    MySqlSourceEncryption(sslMode = MySqlSourceEncryption.SslMode.PREFERRED)
-                }
-                is EncryptionRequired ->
-                    MySqlSourceEncryption(sslMode = MySqlSourceEncryption.SslMode.REQUIRED)
-                is SslVerifyCertificate ->
-                    MySqlSourceEncryption(
-                        sslMode = MySqlSourceEncryption.SslMode.VERIFY_CA,
-                        caCertificate = encryption.sslCertificate,
-                        clientCertificate = encryption.sslClientCertificate,
-                        clientKey = encryption.sslClientKey,
-                        clientKeyPassword = encryption.sslClientPassword
-                    )
-                is SslVerifyIdentity ->
-                    MySqlSourceEncryption(
-                        sslMode = MySqlSourceEncryption.SslMode.VERIFY_IDENTITY,
-                        caCertificate = encryption.sslCertificate,
-                        clientCertificate = encryption.sslClientCertificate,
-                        clientKey = encryption.sslClientKey,
-                        clientKeyPassword = encryption.sslClientPassword
-                    )
-                null -> TODO()
-            }
-        val sslJdbcParameters = jdbcEncryption.parseSSLConfig()
-        jdbcProperties.putAll(sslJdbcParameters)
-
-        val cursorConfig = pojo.getCursorMethodConfigurationValue()
-        val maxSnapshotReadTime: Duration? =
-            when (cursorConfig is CdcCursor) {
-                true -> cursorConfig.initialLoadTimeoutHours?.let { Duration.ofHours(it.toLong()) }
-                else -> null
-            }
-        // Build JDBC URL
+
+        // Configure SSH tunneling.
+        val sshTunnel: SshTunnelMethodConfiguration? = pojo.getTunnelMethodValue()
+        val sshOpts: SshConnectionOptions =
+            SshConnectionOptions.fromAdditionalProperties(pojo.getAdditionalProperties())
+
+        // Configure SSL encryption.
+        if (
+            pojo.getEncryptionValue() is EncryptionPreferred &&
+                sshTunnel is SshNoTunnelMethod &&
+                featureFlags.contains(FeatureFlag.AIRBYTE_CLOUD_DEPLOYMENT)
+        ) {
+            throw ConfigErrorException(
+                "Connection from Airbyte Cloud requires SSL encryption or an SSH tunnel."
+            )
+        }
+        val sslJdbcProperties: Map<String, String> = fromEncryptionSpec(pojo.getEncryptionValue()!!)
+        jdbcProperties.putAll(sslJdbcProperties)
```

Comment:
A question off topic: how do we test the tunneling and encryption that's working?

Reply:
Tunneling is implemented in the CDK which includes tests for it involving docker containers (search for "bastion"). Encryption testing is the responsibility of the connector, since that's handled at the JDBC driver level. From what I can tell, this connector is missing junit tests to that effect, which is a mistake. Perhaps we test this in the CAT suite. I will follow up on this.